src/main/kotlin/org/rust/ide/documentation/RustDocumentationProvider.kt

```diff
@@ -2,12 +2,43 @@ package org.rust.ide.documentation
 
 import com.intellij.lang.documentation.AbstractDocumentationProvider
 import com.intellij.psi.PsiElement
-import org.rust.lang.core.psi.RustFnItem
-import org.rust.lang.core.psi.RustPatBinding
+import com.intellij.util.text.MarkdownUtil
+import com.petebevin.markdown.MarkdownProcessor
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.impl.RustImplMethodMemberImpl
 import org.rust.lang.core.psi.impl.mixin.isMut
 
 class RustDocumentationProvider : AbstractDocumentationProvider() {
 
+    override fun generateDoc(element: PsiElement?, originalElement: PsiElement?): String? {
+        when (element) {
+            is RustItem -> {
+                val doc = element.documentation
+                return if (doc != null) formatDoc(element.name!!, doc)
```

Comment:
It is a bad idea to use `!!`.

Name of an `item` can be null because the parser recovers aggressively (at least, it should).

If you have a syntax error in your code 

```
struct {

}
``` 

Then, as soon as the parser sees the `struct` keyword, it will generate a `RustStructItem` PSI node, but it's name will be `null`.